rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -195,20 +212,13 @@ object Reduce {
             else
               rand.splitByte((starts(5) + idx).toByte)
 
-          expr.exprInstance match {
-            case EVarBody(EVar(v)) =>
-              (for {
-                varref <- eval(v)
-                _      <- eval(varref)(env, newRand, costAccountingAlg)
-              } yield ()).handleError(fTell.tell)
-            case e: EMethodBody =>
-              (for {
-                p <- evalExprToPar(Expr(e))
-                _ <- eval(p)(env, newRand, costAccountingAlg)
-              } yield ()).handleError(fTell.tell)
-            case _ => s.unit
-          }
+          exprHandler(expr, newRand)
         })
+        Parallel.parTraverse(par.sends.zipWithIndex.toList)(handle(mkTermHandler(evalExplicit), starts(0))),
```

Comment:
Please retain the original order of traverses this and the lines 217-221 should be above line 215

Reply:
I don't understand what you mean here

Reply:
This doesn't really matter because we traverse it in parallel.